### ISSUE TITLE
Failure to run flood on buildings using IF Wizard vers 4

### ISSUE THREAD

**Issue by NyakudyaA** [inasafe/inasafe#3741]
### Problem
When running flood on buildings using wizard there is an error that comes.

![out](https://cloud.githubusercontent.com/assets/2510900/22456585/10a4a4a8-e79c-11e6-908f-b01e821b189b.gif)

```
An error has occurred while executing Python code: 

AttributeError: 'StepFcAnalysis' object has no attribute 'extent' 
Traceback (most recent call last):
  File "/home/admire/.qgis2/python/plugins/inasafe/safe/gui/tools/wizard/wizard_dialog.py", line 656, in on_pbnNext_released
    self.go_to_step(new_step)
  File "/home/admire/.qgis2/python/plugins/inasafe/safe/gui/tools/wizard/wizard_dialog.py", line 592, in go_to_step
    self.step_fc_analysis.setup_and_run_analysis()
  File "/home/admire/.qgis2/python/plugins/inasafe/safe/gui/tools/wizard/step_fc90_analysis.py", line 114, in setup_and_run_analysis
    self.impact_function = self.prepare_impact_function()
  File "/home/admire/.qgis2/python/plugins/inasafe/safe/gui/tools/wizard/step_fc90_analysis.py", line 188, in prepare_impact_function
    if self.extent.user_extent:
AttributeError: 'StepFcAnalysis' object has no attribute 'extent'


Python version: 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609] 
QGIS version: 2.18.2 Las Palmas, 102ee87 

```

### Expected Behaviour
Run analysis without errors

### Environment
* InaSAFE :  develop
* QGIS : 2.18.3 and 2.14.3
* OS :  Ubuntu 16.04
* Data used : `safe/test/data`
  * Exposure :  `exposure/buildings.shp`
  * Hazard : `hazard/floods.shp`
  * Aggregation : None

### Proposed Solution
Run analysis

**Comment by Gustry** [comment]
Thanks, I got this one too

**Comment by ismailsunni** [comment]
I got this one also today. Strange, may be I break the code somewhere, it was working before. Ouch, we really need a better unit test.

**Comment by NyakudyaA** [comment]
@Gustry , @ismailsunni  It seems most of the IF wizard is broken. Have just tested with Ash raster on building centroids and have the same error as above.

![ifc](https://cloud.githubusercontent.com/assets/2510900/22463996/67157710-e7bd-11e6-81d8-f201165ee0f1.png)

**Comment by Gustry** [comment]
I would say, in inasafe v4, we have only one IF ;-)
And this bug is in the wizard, so it will happen whatever the layer you use.

**Comment by ismailsunni** [comment]
Yes, it's the IFCW that cause the bug.

**Comment by Gustry** [comment]
Also as  I reminder, the IF is now returning more errors with the `prepare` that the wizard also need to catch : https://github.com/inasafe/inasafe/blob/develop/safe/definitions/constants.py#L42

I think we should make some groups.

**Comment by myarjunar** [comment]
closed in #3805
